c7n/policy.py

```diff
@@ -700,12 +701,35 @@ def is_lambda(self):
         return True
 
     def validate(self):
+        self._check_deprecation('resource', self.resource_manager)
+
         m = self.get_execution_mode()
         m.validate()
+
+        parent_element_name = self.resource_manager.type
```

Comment:
resource types aren't really parent elements, their just types. elements are really filters and actions, so a parent element is typically a `or` or `and` clause when nesting.